mazeRunner/src/mazeRunner.js

```diff
@@ -1,68 +1,70 @@
-  const newPosition = (direction, position) => {
-    switch (direction) {
-      case "E":
-        return [position[0], position[1] + 1];
-      case "W":
-        return [position[0], position[1] - 1];
-      case "N":
-        return [position[0] - 1, position[1]];
-      case "S":
-        return [position[0] + 1, position[1]];
+  // takes a direction (string) and position (tuple) and returns new position (tuple)
+  newPosition = (direction, position) => {
+    if (Array.isArray(position) && position.length === 2) {
+      switch (direction) {
+        case "E":
+          return [position[0], position[1] + 1];
+        case "W":
+          return [position[0], position[1] - 1];
+        case "N":
+          return [position[0] - 1, position[1]];
+        case "S":
+          return [position[0] + 1, position[1]];
+      }
+    } else {
+      throw new Error("Position must be a tuple");
     }
   };
 
-  //console.log("original position of the 2", startingPosition(maze));
+  // calculates the outcome at the maze position
+  result = (position) => {
+    const mazePoint = this.maze[position[0]][position[1]];
 
-  const result = (maze, position) => {
-    // console.log("position in result", position);
-    // console.log("maze point", maze[position[0]][position[1]]);
-    if (
-      maze[position[0]][position[1]] === 0 ||
-      maze[position[0]][position[1]] === 2
-    ) {
+    if (mazePoint === 0 || mazePoint === 2) {
       return "Lost";
-    } else if (maze[position[0]][position[1]] === 1) {
+    } else if (mazePoint === 1) {
       return "Dead";
-    } else if (maze[position[0]][position[1]] === 3) {
+    } else if (mazePoint === 3) {
       return "Finish";
     }
   };
 
-  for (let i = 0; i < directions.length; i++) {
-    if (i === 0) {
-      position = startingPosition(maze);
-    }
+  attemptWith = (directions) => {
+    let position = this.startingPosition();
+    let moveNumber = 0;
 
-    position = newPosition(directions[i], position);
-    //console.log("position in for loop", position);
-    if (position.includes(maze.length) || position.includes(-1)) {
-      return "Dead";
-    }
-    outcome = result(maze, position);
-    //console.log("outcome at new position", outcome);
-    // console.log("i", i + 1);
-    // console.log("number of directions", directions.length);
-    if (outcome === "Dead" || outcome === "Finish") {
-      return outcome;
-    }
-    if (outcome === "Lost" && i + 1 === directions.length) {
-      return "Lost";
+    for (const direction of directions) {
+      position = this.newPosition(direction, position);
+      moveNumber++;
+      if (position.includes(this.maze.length) || position.includes(-1)) {
+        return "Dead";
+      }
+      let outcome = this.result(position);
+
+      if (outcome === "Dead" || outcome === "Finish") {
+        return outcome;
+      } else if (outcome === "Lost" && moveNumber === directions.length) {
+        return "Lost";
+      }
     }
```

Comment:
Having the `moveNumber` variable is you getting around the `for..of` loop not having which iteration it's on as a variable (like you had before with `i`). Again, that's my fault for suggesting a `for...of` loop. Sorry!

I should say what can be the problem with doing the `for (let i = 0; i < 5; i++)` style loops is. It's very easy to get off by 1 errors, where the iteration and array index don't line up.

In this case having the `moveNumber` variable actually makes me have to go through the same thought process. My thought process is:
'oh we have a variable that's getting incremented in each loop'
'oh it's being used in the condition to check whether the move is the last one'
'oh wait this variable started at 0, and is being checked against the length of the array'
'I know it's easy to end one iteration too early as counting array length starts at 1 rather than 0 (e.g. a one element array has length 1, but in the first iteration `moveNumber` will start at 0)'
'ok so I have to make sure that `moveNumber` is always incremented before I get to the check of the array length'

A slight vision in my thoughts....

However, with the `for...in` loop, we know that it's going to run for each of the elements in the array (as long as it doesn't return), so we don't have to keep track of if it's the last element. In theory we could move that check out of the loop, with the presumption that if it finishes the loop then it has done all the moves. More concretely, I think you could do:
```
for (const direction of directions) {
 .....
}
if (this.result(position) === "Lost") {
    return "Lost";
}
```
I think technically you might not even need the if condition and you could just do:
```
for (const direction of directions) {
 .....
}
return "Lost";
```
As, if you didn't do any of the other things in your moves then by default you're lost?